### PR TITLE
Implement backend unit tests [LVNODJ-109] [Oleksii]

### PR DESCRIPTION
**Pull request by vglfr**

### CODE REVIEW

File: server/src/helpers/test/validation.test.ts [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+import mockingoose from 'mockingoose';
+
+describe('Validate registration form helper:', () => {
+    test('validate input - failure (no fields provided)', async () => {
+        Validator.isEmpty = jest.fn(() => true);
+        Validator.isEmail = jest.fn(() => true);
+        Validator.isLength = jest.fn(() => true);
+        jest.mock('lodash/isEmpty', () => false);
+
+        const inp = {};
```
Comment (InnaKutsik) [inline server/src/helpers/test/validation.test.ts]: pls use full names - input

File: server/src/helpers/test/validation.test.ts [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        const out = {
```
Comment (InnaKutsik) [inline server/src/helpers/test/validation.test.ts]: pls use full names - output

File: server/src/helpers/test/validation.test.ts [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        expect(validateInput(fakeUser as IUserModel)).toStrictEqual(out);
+    });
+
+    test('validate input - failure (invalid email and password fields provided)', async () => {
+        Validator.isEmpty = jest.fn(() => false);
+        Validator.isEmail = jest.fn(() => false);
+        Validator.isLength = jest.fn(() => false);
+        jest.mock('lodash/isEmpty', () => false);
+
+        const inp = {
```
Comment (InnaKutsik) [inline server/src/helpers/test/validation.test.ts]: pls use full names - input

File: server/src/helpers/test/validation.test.ts [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+            email: 'in@valid',
+            fillName: 'valid',
+            username: 'valid',
+            password: 'invalid',
+        };
+        const out = {
```
Comment (InnaKutsik) [inline server/src/helpers/test/validation.test.ts]: pls use full names - output

File: server/src/helpers/test/validation.test.ts [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+        Validator.isLength = jest.fn(() => true);
+        jest.mock('lodash/isEmpty', () => true);
+
+        const inp = {
+            email: 'valid@ema.il',
+            fillName: 'valid',
+            username: 'valid',
+            password: 'longandvalid',
+        };
+        const out = {
```
Comment (InnaKutsik) [inline server/src/helpers/test/validation.test.ts]: pls use full names - output

File: server/src/helpers/test/validation.test.ts [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+        expect(validateInput(fakeUser as IUserModel)).toStrictEqual(out);
+    });
+
+    test('validate input - success', async () => {
+        Validator.isEmpty = jest.fn(() => false);
+        Validator.isEmail = jest.fn(() => true);
+        Validator.isLength = jest.fn(() => true);
+        jest.mock('lodash/isEmpty', () => true);
+
+        const inp = {
```
Comment (InnaKutsik) [inline server/src/helpers/test/validation.test.ts]: pls use full names - input

**Comment by vglfr** [comment]
Coverage:
- 7 tests suites skipped, 15 passed, total 22
- 25 tests skipped, 28 passed, total 53

File: server/src/helpers/test/validation.test.ts (hunk 1 above)
Comment (vglfr) [reply server/src/helpers/test/validation.test.ts]: fixed

File: server/src/helpers/test/validation.test.ts (hunk 2 above)
Comment (vglfr) [reply server/src/helpers/test/validation.test.ts]: fixed

File: server/src/helpers/test/validation.test.ts (hunk 3 above)
Comment (vglfr) [reply server/src/helpers/test/validation.test.ts]: fixed

File: server/src/helpers/test/validation.test.ts (hunk 4 above)
Comment (vglfr) [reply server/src/helpers/test/validation.test.ts]: fixed

File: server/src/helpers/test/validation.test.ts (hunk 5 above)
Comment (vglfr) [reply server/src/helpers/test/validation.test.ts]: fixed

File: server/src/helpers/test/validation.test.ts (hunk 6 above)
Comment (vglfr) [reply server/src/helpers/test/validation.test.ts]: fixed

**Comment by vglfr** [comment]
Short names (inp, out, ans) grepped out and fixed.

**Pull request merged**